crates/bevy_asset/src/io/wasm.rs

```diff
@@ -38,8 +38,27 @@ fn js_value_to_err<'a>(context: &'a str) -> impl FnOnce(JsValue) -> std::io::Err
 
 impl HttpWasmAssetReader {
     async fn fetch_bytes<'a>(&self, path: PathBuf) -> Result<Box<Reader<'a>>, AssetReaderError> {
-        let window = web_sys::window().unwrap();
-        let resp_value = JsFuture::from(window.fetch_with_str(path.to_str().unwrap()))
+        let global = js_sys::global();
+        let maybe_window = Reflect::get(&global, &JsValue::from_str("Window"))
```

Comment:
This seems unnecessarily "stringy". Is there a reason to not do this instead?

```rust
let global: Global = js_sys::global().unchecked_into();

if !global.window().is_undefined() {
    let window: Window = global.unchecked_into();
} else if !global.worker().is_undefined() {
    let worker: WorkerGlobalScope = global.unchecked_into();
} 
```

Reply:
Ahhhh because we don't have a Global type

Reply:
That was defined in the project I got the snippet from.

Reply:
In theory we could add it ourselves / generate it via wasm_bindgen:

https://github.com/PWhiddy/wgpu-hackdown/blob/d7296ac30b7948d6d111ffe201d7c47c246d16cd/wgpu/src/backend/web.rs#L993C1-L1002C2

Reply:
Generating the bindings _feels_ like it would be more efficient, but I don't have the expertise here to say with certainty.

Reply:
We could add a Global type, which project did you get it from? The PR here is based on the [solution used in gloo-net](https://github.com/rustwasm/gloo/blob/master/crates/net/src/http/request.rs#L323-L338).

Reply:
> Generating the bindings feels like it would be more efficient

I was thinking about that but then I realised that the next step is to do a HTTP fetch which is probably 100000x slower than anything involving converting strings back and forth between Js and Rust...

Reply:
@daxpedda any thoughts?

Reply:
> > Generating the bindings feels like it would be more efficient
> 
> I was thinking about that but then I realised that the next step is to do a HTTP fetch which is probably 100000x slower than anything involving converting strings back and forth between Js and Rust...

True.

> @daxpedda any thoughts?

Generating custom bindings is the way this is usually done, it would probably make more sense Bevy had a centralized place to make these bindings if they are used elsewhere.

So my 2¢: if this is the only occurrence, leave it as-is, if not, make bindings like suggested in https://github.com/bevyengine/bevy/pull/12134/files#r1505095751.

Reply:
I feel this code almost belongs in wasm-bindgen itself, so far some variant of this logic has been replicated in gloo-timers, gloo-net, wgpu, and now we considering putting it into bevy_assets.

I do not have a strong opinion but I am happy to create the relevants PRs. The two open questions are then:
1. Should we add a binding for the `Global` object (I feel like we are leaning towards "yes")
2. Where should that binding go? (1) wasm.rs at file scope, (2) wasm.rs at function scope (inside fetch_bytes), (3) somewhere else in Bevy, (4) in wasm-bindgen.

Reply:
> I feel this code almost belongs in wasm-bindgen itself, so far some variant of this logic has been replicated in gloo-timers, gloo-net, wgpu, and now we are considering putting it into bevy_assets.

I don't believe this should be done in `wasm-bindgen`, as its only purpose is to provide bindings and nothing else. It might be more appropriate in `gloo`. That said, `wasm-bindgen` needs a general mechanism to query the availability of a binding, which would probably solve this problem as well.

Reply:
> I don't believe this should be done in `wasm-bindgen`

Indeed, I meant to say inside of `js_sys`

Reply:
> > I don't believe this should be done in `wasm-bindgen`
> 
> Indeed, I meant to say inside of `js_sys`

Same answer really, `js-sys` is also bindings only. But a general purpose solution to this problem would probably live in `web-sys`? Probably off-topic to discuss here.